jobs/microk8s/charms/release.sh

```diff
@@ -7,72 +7,57 @@
+
+## default optional variables
+JOB_NAME="${JOB_NAME:-unnamed-job}"
+BUILD_NUMBER="${BUILD_NUMBER:-0}"
+export JUJU_MODEL=${JOB_NAME}-${BUILD_NUMBER}
+
 # Cleanup old containers
-container_prefix="${JOB_NAME}"
-old_containers=$(sudo lxc list -c n -f csv "${container_prefix}" | xargs)
-if [[ ! -z $old_containers ]]; then
-  echo Removing old containers, $old_containers
-  sudo lxc delete --force $old_containers
-fi
+ci_lxc_delete "${JOB_NAME}"
 
 # Configure cleanup routine
-container="${container_prefix}-${BUILD_NUMBER}"
+export charmcraft_lxc="${JOB_NAME}-${BUILD_NUMBER}"
 function cleanup() {
   set +e
-  sudo lxc shell $container -- bash -c 'charmcraft logout'
-  sudo lxc delete $container --force
+  ci_lxc_delete $charmcraft_lxc
   juju destroy-model $JUJU_MODEL -y --timeout 1m --force
   rm -rf $JUJU_DATA
   set -e
 }
 trap cleanup EXIT
 
-# Launch local LXD container to publish to charmcraft
-sudo lxc launch ubuntu:20.04 $container
-timeout 5m bash -c "
-  until sudo lxc shell $container -- bash -c 'snap install charmcraft --classic'; do
-    sleep 3
-  done
-"
-
-# TODO: update when charmcraft supports non-interactive logins
-# sudo lxc file push $CHARMCRAFT_CREDENTIALS $container/charmcraft.credentials
-# sudo lxc shell $container -- bash -c 'mkdir -p ~/snap/charmcraft/common/config/charmcraft.credentials'
-# sudo lxc shell $container -- bash -c 'cp charmcraft.credentials ~/snap/charmcraft/common/config/charmcraft.credentials'
-sudo lxc shell $container -- bash -c 'charmcraft login'
-sudo lxc shell $container -- bash -c 'charmcraft whoami'
-
-# Retrieve revision from CharmHub API. For reference:
+# Launch container
+ci_charmcraft_launch $charmcraft_lxc
+
+# Retrieve revision from CharmHub API. For reference (http://api.snapcraft.io/docs/charms.html#charm_info). Note that `head -1` is needed because the info endpoint returns an entry for each base.
 #
-# curl --silent https://api.charmhub.io/v1/charm/microk8s-testing/releases -b ./snap/charmcraft/common/config/charmcraft.credentials  | jq '.["channel-map"][] | {channel: .channel, revision: .revision}'
-# {
-#   "channel": "latest/edge",
-#   "revision": 9
-# }
-# {
-#   "channel": "latest/stable",
-#   "revision": 3
-# }
-revision=$(sudo lxc shell $container -- \
-  bash -c "curl https://api.charmhub.io/v1/charm/$CHARM_NAME/releases -b ~/snap/charmcraft/common/config/charmcraft.credentials" \
-    | jq -r ".[\"channel-map\"][] | select(.channel == \"$FROM_CHANNEL\") | .revision")
+# curl 'https://api.charmhub.io/v2/charms/info/microk8s?fields=channel-map' | jq -r '.["channel-map"][] | select(.channel.name == "edge") | select(.channel.base.architecture == "amd64") | .revision.revision' | head -1
+# 20
+revision=$(curl --silent 'https://api.charmhub.io/v2/charms/info/microk8s?fields=channel-map' | jq -r '.["channel-map"][] | select(.channel.name == "'"$FROM_CHANNEL"'") | select(.channel.base.architecture == "amd64") | .revision.revision' | head -1)
```

Comment:
this looks like fun.  i'm guessing there's not a `charmcraft` command for this?

something like [this](https://github.com/charmed-kubernetes/jenkins/blob/main/jobs/build-charms/charms.py#L218-L222)

```bash
$ charmcraft revisions microk8s
```

isn't fun to parse the table'd output

Reply:
> isn't fun to parse the table'd output

if it had a parsable output option, i would be much happier. it's challenging because of the multiple lines and architectures getting in the way.

Reply:
https://github.com/canonical/charmcraft/issues/183